### ISSUE TITLE
Output rules in a specific format

### ISSUE THREAD

**Issue by RickvdStaaij** [particle-php/Validator#80]
Create an output function on the validator, that returns a ruleset in a desired output format.

After you configured the rules in the Validator, you might want to output the rules to a specific format. This is usefull when you want to have the same validation on your API and GUI, or the same validation in your front-end as you PHP for example.

Desired validator API:
```php
use Particle\Validator\Validator;
use Particle\Validator\Format;

$v = new Validator();

$v->required('first_name')->lengthBetween(2, 30);

/**
 * $ruleSet = [
 *     'first_name' => [
 *         [
 *             'name' => 'LengthBetween',
 *             'options' => [
 *                 'min' => 2,
 *                 'max' => 30,
 *             ],
 *             'messages' => [
 *                 'tooLong' => 'Your first_name may not be longer than 30 characters',
 *                 'tooShort' => 'Your first_name must be at least 2 characters long',
 *             ],
 *         ],
 *     ],
 * ]
 */
$customRules = $v->output(function(array $ruleSet) {
    return json_encode($ruleSet);
});
```

**Comment by berry-langerak** [comment]
Hi! The way I had imagined this functionality was that you could *only* pass in a `Callable` into output (which will receive a dumb tree of Transfer Objects). I really dislike the idea of adding converting to JSON or YAML, or XML into the library itself; we can create new repositories for doing that. Let's try to keep Validator itself as small as possible.

**Comment by RickvdStaaij** [comment]
Yes fair enough, I updated the post and I added $messages to the resulted rule set.

**Comment by berry-langerak** [comment]
Excellent :)

**Comment by RickvdStaaij** [comment]
This is completed in #82, and released in `v2.0.0-alpha.2` in preparation of `v2.0.0`.
